bwpy/mea_viewer.py

```diff
@@ -0,0 +1,77 @@
+from . import signal
+import abc
+import plotly.graph_objects as go
+import numpy as np
+from scipy.stats import norm
+
+
+class Viewer(abc.ABC):
+    @abc.abstractmethod
+    def build_view(self, slice, bin_size, view_method):
+        pass
+
+
+class MEAViewer(Viewer):
+    colorscale = [[0, "#ebe834"], [1.0, "#eb4034"]]
+    min_value = 0
+    max_value = 0
+
+    def build_view(
+        self, file, slice=None, view_method="amplitude", bin_size=100, data=None
+    ):
+        fig = go.Figure()
+        if slice and data:
+            raise ValueError("slice and data arguments are mutually exclusives.")
+        if slice:
+            apply_transformation = getattr(signal, view_method)
+            signals = apply_transformation(slice, bin_size).data
+        else:
+            signals = data
+
+        for signal_frame in signals:
+            fig.add_trace(
+                go.Heatmap(
+                    visible=False,
+                    z=signal_frame,
+                    zmin=self.min_value,
+                    zmax=self.get_up_bound(signals, file),
```

Comment:
This is constant for the entire loop, please hoist this expression out of the loop.